runtime/memprof.c

```diff
@@ -2281,6 +2287,42 @@ CAMLprim value caml_memprof_start(value lv, value szv, value tracker)
   CAMLreturn(config);
 }
 
+CAMLprim value caml_memprof_participate(value config)
+{
```

Comment:
I think it's probably fine as is, but it'd be more obviously fine if this function used CAMLparam/CAMLreturn.